docs/getting_started/contribution_guide.rst

```diff
@@ -59,4 +59,120 @@ Kiso-testing and make sure that you understand your rights and obligations.
 Any contributions to Kiso-testing project code base needs to be licensed under
 EPL v2.0.
 
+What should I do before committing ?
+------------------------------------
+
+PEP8-Compliancy
+---------------
+In order to maintain clear and user-friendly project make sure, that your
+changes respect PEP8 standards. PEP8 is a guide that provides Python best
+practices (naming, indentation,...).
+Official document : https://peps.python.org/pep-0008/
+
+To make sure your changes are PEP8 Compliant it exists different tools, that
+help you on this tasks :
+    - linter (applicable on IDE) :
+        show some warning directly on IDE
+    - pre-commit hook :
+        hook scripts that look after different issues (white space, trailing,...)
+
+
+Typography
+----------
+Most of the comments made during PR-Reviewing are about typography/misspelling
+mistakes. An easy way to avoid these is by running
+`codespell <https://pypi.org/project/codespell/>`_ on your written code.
+
+
+Function Annotation
+-------------------
+Inside the ITF, every implemented function must have annotations for its
+parameters and return types. This results in increased readability and
+therefore in easier comprehension of the code for any reader.
+
+    .. code:: python
+
+        def some_fun(some_dict_param: dict, some_string_param: str) -> list:
+
+    .. note:: As not every types are available in the builtins, or as it is
+        important to precise inner type you might import some from collections
+        module or typing
+
+    .. code:: python
+
+        from typing import List
+        from collections import namedtuple
+
+        def some_fun(
+            some_int_list_param: List[int], some_imported_type_param: namedtuple
+        ) -> list:
+
+
+Testing
+-------
+To ensure the correct behaviour of your code, add unit tests for every of
+your implemented functions. A convenient and pythonic way to do this is
+given by `pytest <https://docs.pytest.org/en/stable/>`_
+Unittesting is ensure with `codecov <https://docs.codecov.com/docs>`_. It
+simply checks code coverage is not going lower than it was before changes.
+
+To ensure proper integration of your changes into the existing features,
+and demonstrate their usages, adapt the examples of modified module, and run
```

Comment:
I would put a different section for the examples :)

Reply:
Do you mean that, I should split Testing section into 2 differents ?